addons/point_of_sale/models/pos_session.py

```diff
@@ -756,9 +756,11 @@ def _accumulate_amounts(self, data):
                         line['base_tags'],
                     )
                     sales[sale_key] = self._update_amounts(sales[sale_key], {'amount': line['amount']}, line['date_order'])
+                    sales[sale_key]['tax_amount'] = 0
```

Comment:
I think we should conditionally set this to zero. It's possible that the same "sale_key" is generated from different orderlines. I think we can do this:

```suggestion
                    sales[sale_key].setdefault('tax_amount', 0)
```

And in the test, create second product with the same tax as the first product. Sell and check if the values are correct.